core/API/DataTablePostProcessor.php

```diff
@@ -421,16 +430,68 @@ public function computeProcessedMetrics(DataTable $dataTable)
             }
 
             foreach ($dataTable->getRows() as $row) {
-                if ($row->getColumn($name) === false) { // only compute the metric if it has not been computed already
-                    $computedValue = $processedMetric->compute($row);
-                    if ($computedValue !== false) {
-                        $row->addColumn($name, $computedValue);
-                    }
-
-                    $subtable = $row->getSubtable();
-                    if (!empty($subtable)) {
-                        $this->computeProcessedMetrics($subtable);
-                    }
+                if ($row->getColumn($name) !== false) { // only compute the metric if it has not been computed already
+                    continue;
+                }
+
+                $computedValue = $processedMetric->compute($row);
+                if ($computedValue !== false) {
+                    $row->addColumn($name, $computedValue);
+                }
+            }
+        }
+
+        foreach ($dataTable->getRows() as $row) {
```

Comment:
can you comment here why this is needed and what it does? 

Also can we check this for some bigger subtables re performance etc?

Reply:
Note: this is one of several issues I fixed while doing this work. Before this loop was executed for every processed metric being computed, instead of once for the whole table. So it was doing extra work.